dsgrid/registry/registry_manager.py

```diff
@@ -221,6 +222,97 @@ def load(
+                config_file, self.dimension_manager, self.dimension_mapping_manager
+            )
+            if dataset_id:
+                if dataset_id not in config.list_registered_dataset_ids():
+                    raise ValueError(
+                        f"No registered dataset ID = '{dataset_id}' registered to project ID = '{project_id}'"
+                    )
+                datasets = [(dataset_id, str(config.get_dataset(dataset_id).version))]
+            else:
+                datasets = []
+                for dataset in config.list_registered_dataset_ids():
+                    datasets.append((dataset, str(config.get_dataset(dataset).version)))
+
+        if dataset_id and not project_id:
+            if not self.dataset_manager.has_id(dataset_id):
+                raise ValueError(f"No registered dataset ID = '{dataset_id}'")
+            version = self.dataset_manager.get_current_version(dataset_id)
+            datasets = [(dataset_id, version)]
+
+        for dataset, version in datasets:
+            self._data_sync(dataset, version, no_prompts)
+
+    def _data_sync(self, dataset_id, version, no_prompts=True):
+        cloud_interface = self._params.cloud_interface
+        offline_mode = self._params.offline
+        dry_run_mode = self._params.dry_run
+
+        if offline_mode:
+            raise ValueError("dsgrid data-sync only works in online mode.")
+        if dry_run_mode == True:
+            sync = False
+        else:
+            sync = True
+
+        lock_files = list(
+            cloud_interface.get_lock_files(
+                relative_path=f"s3:/{cloud_interface._s3_filesystem._bucket}/configs/datasets/{dataset_id}"
+            )
+        )
+        if lock_files:
+            msg = f"There are {len(lock_files)} lock files in the registry:"
+            for lock_file in lock_files:
+                msg = msg + "\n\t" + f"- {lock_file}"
+            logger.log(msg)
+            if not no_prompts:
+                msg = msg + "\n... Do you want to continue syncing the registry contents? [Y] >>> "
+                val = input(msg)
+                if val == "" or val.lower() == "y":
+                    sync = True
+                else:
+                    logger.info("Skipping remote registry sync.")
+
+        if sync:
+            logger.info(f"Sync data from remote registry for {dataset_id}, version={version}.")
+            cloud_interface.sync_pull(
+                remote_path=self._params.remote_path + f"/data/{dataset_id}/{version}",
+                local_path=str(self._params.base_path) + f"/data/{dataset_id}/{version}",
+                exclude=SYNC_EXCLUDE_LIST,
+                delete_local=True,
+            )
+            cloud_interface.sync_pull(
+                remote_path=self._params.remote_path + f"/data/{dataset_id}/registry.toml",
+                local_path=str(self._params.base_path) + f"/data/{dataset_id}/registry.toml",
+                exclude=SYNC_EXCLUDE_LIST,
```

Comment:
This isn't needed, right?

Reply:
the first sync_pull uses the aws sync command to sync directories and files. The second one is for a single file which uses the aws cp command. Both are needed.

Reply:
I was referring to this specific line:
```
exclude=SYNC_EXCLUDE_LIST,
```